pkg/client/cli/cloud_messages.go

```diff
@@ -0,0 +1,157 @@
+package cli
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	empty "google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/spf13/cobra"
+
+	"github.com/datawire/dlib/dtime"
+
+	"github.com/telepresenceio/telepresence/rpc/v2/systema"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cache"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/cliutil"
+)
+
+const messagesCacheFilename = "cloud-messages.json"
+
+type cloudMessageCache struct {
+	NextCheck         time.Time       `json:"next_check"`
+	Intercept         string          `json:"intercept"`
+	MessagesDelivered map[string]bool `json:"messagess_delivered"`
```

Comment:
hmm it seems like this could easily be a `map[string]struct{}` and instead of setting the value to false the key could just be deleted. I guess this way lets us distinguish between a message that's never been delivered and one that has changed, but it doesn't look like that's something we need right now

Reply:
Good point, I made this more complicated than it needed to be!